Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -592,159 +592,184 @@ public actor SourceKitServer {
-        await self.handleRequest(for: request, requestHandler: self.documentSymbolHighlight, fallback: nil)
-      case let request as Request<FoldingRangeRequest>:
-        await self.handleRequest(for: request, requestHandler: self.foldingRange, fallback: nil)
-      case let request as Request<DocumentSymbolRequest>:
-        await self.handleRequest(for: request, requestHandler: self.documentSymbol, fallback: nil)
-      case let request as Request<DocumentColorRequest>:
-        await self.handleRequest(for: request, requestHandler: self.documentColor, fallback: [])
-      case let request as Request<DocumentSemanticTokensRequest>:
-        await self.handleRequest(for: request, requestHandler: self.documentSemanticTokens, fallback: nil)
-      case let request as Request<DocumentSemanticTokensDeltaRequest>:
-        await self.handleRequest(for: request, requestHandler: self.documentSemanticTokensDelta, fallback: nil)
-      case let request as Request<DocumentSemanticTokensRangeRequest>:
-        await self.handleRequest(for: request, requestHandler: self.documentSemanticTokensRange, fallback: nil)
-      case let request as Request<ColorPresentationRequest>:
-        await self.handleRequest(for: request, requestHandler: self.colorPresentation, fallback: [])
-      case let request as Request<CodeActionRequest>:
-        await self.handleRequest(for: request, requestHandler: self.codeAction, fallback: nil)
-      case let request as Request<InlayHintRequest>:
-        await self.handleRequest(for: request, requestHandler: self.inlayHint, fallback: [])
-      case let request as Request<DocumentDiagnosticsRequest>:
-        await self.handleRequest(
-          for: request,
-          requestHandler: self.documentDiagnostic,
-          fallback: .full(.init(items: []))
-        )
-        // IMPORTANT: When adding a new entry to this switch, also add it to the `TaskMetadata` initializer.
-      default:
-        reply(.failure(ResponseError.methodNotFound(R.method)))
+      await withLoggingScope("request-\(id)") {
+        await self.handleImpl(params, id: id, from: clientID, reply: reply)
       }
     }
   }
 
-  private nonisolated func _logRequest<R>(_ request: Request<R>) {
-    logger.log("Received request: \(request.forLogging)")
-  }
+  private func handleImpl<R: RequestType>(
+    _ params: R,
+    id: RequestID,
+    from clientID: ObjectIdentifier,
+    reply: @escaping (LSPResult<R.Response>) -> Void
+  ) async {
+    let startDate = Date()
+
+    let cancellationToken = CancellationToken()
+
+    let request = Request(
+      params,
+      id: id,
+      clientID: clientID,
+      cancellation: cancellationToken,
+      reply: { [weak self] result in
+        reply(result)
+        let endDate = Date()
+        if let self {
+          Task {
+            logger.debug(
+              """
+              Sending response (took \(endDate.timeIntervalSince(startDate) * 1000)ms):
```

Comment:
Same comment as usual with the `:` but super nitpicky and I don't really care, we can change at any time.

Reply:
Changed it to just be a newline